core/domain/user_jobs_one_off_test.py

```diff
@@ -1749,6 +1584,95 @@ def test_one_setting_model_without_gae_id(self):
             migrated_setting_model.last_updated)
 
 
+class MockUserSubscriptionsModelWithActivityIDs(
+        user_models.UserSubscriptionsModel):
+    """Mock UserSubscriptionsModel so that it allows to set 'activity_ids'. """
+
+    activity_ids = (
+        datastore_services.StringProperty(indexed=True, repeated=True))
+
+
+class RemoveActivityIDsOneOffJobTests(test_utils.GenericTestBase):
+    def _run_one_off_job(self):
+        """Runs the one-off MapReduce job."""
+        job_id = (
+            user_jobs_one_off.RemoveActivityIDsOneOffJob.create_new())
+        user_jobs_one_off.RemoveActivityIDsOneOffJob.enqueue(job_id)
+        self.assertEqual(
+            self.count_jobs_in_mapreduce_taskqueue(
+                taskqueue_services.QUEUE_NAME_ONE_OFF_JOBS), 1)
+        self.process_and_flush_pending_mapreduce_tasks()
+        stringified_output = (
+            user_jobs_one_off.RemoveActivityIDsOneOffJob
+            .get_output(job_id))
+        eval_output = [ast.literal_eval(stringified_item) for
+                       stringified_item in stringified_output]
+        return eval_output
+
+    def test_one_subscription_model_with_activity_ids(self):
+        with self.swap(
+            user_models, 'UserSubscriptionsModel',
+            MockUserSubscriptionsModelWithActivityIDs):
+            original_subscription_model = (
+                user_models.UserSubscriptionsModel(
+                    id='id',
+                    activity_ids=['exp_1', 'exp_2', 'exp_3']
+                )
+            )
+            original_subscription_model.update_timestamps()
+            original_subscription_model.put()
+
+            self.assertIsNotNone(
+                original_subscription_model.activity_ids)
+            self.assertIn(
+                'activity_ids', original_subscription_model._values)  # pylint: disable=protected-access
+            self.assertIn(
+                'activity_ids', original_subscription_model._properties)  # pylint: disable=protected-access
+
+            output = self._run_one_off_job()
+            self.assertItemsEqual(
+                [['SUCCESS_REMOVED - UserSubscriptionsModel', 1]], output)
+
+            migrated_subscription_model = (
+                user_models.UserSubscriptionsModel.get_by_id('id'))
+
+            self.assertNotIn(
+                'activity_ids', migrated_subscription_model._values)  # pylint: disable=protected-access
+            self.assertNotIn(
+                'activity_ids', migrated_subscription_model._properties)  # pylint: disable=protected-access
+            self.assertEqual(
+                original_subscription_model.last_updated,
+                migrated_subscription_model.last_updated)
+
+    def test_one_subscription_model_without_activity_ids(self):
```

Comment:
Add test that will run the job twice, and verify that it works.

Reply:
Done.